backend/users/auth.py

```diff
@@ -10,17 +10,59 @@
 
 from .models import LoginGovUser
 
+from audit.models import Access
+
+import logging
 import string
 import random
 
+logger = logging.getLogger(__name__)
+
+
+def claim_audit_access(user, all_emails):
+    access_invites = (
+        Access.objects.filter(user_id=None)
+        .filter(email__in=all_emails)
+        .update(user_id=user.id)
+    )
+    logger.debug(f"{user.email} granted access to {access_invites} new audits")
+
 
 class JWTUpsertAuthentication(JWTAuthentication):
+    def get_or_create_auth_user(self, email, all_emails):
+        # find all Users where email is in LoginGov all_emails
+        users = self.user_model.objects.filter(email__in=all_emails)
+
+        if users.count() > 1:
+            # more than one found, take the one that matches the LoginGov primary email
+            try:
+                user = users.get(email=email)
+            except self.user_model.DoesNotExist:
+                # none of the Users use the LoginGov primary email, take the first one
+                user = users[0]
+        elif users.count() == 1:
+            # found a single User match
+            user = users[0]
+        else:
+            # didn't find an existing User, create one
+
+            # generate a random password
+            password = "".join(
```

Comment:
Do we want to note here that this password is never going to be used, since our auth is all via login.gov?

Reply:
yes - done!